source/server/config_validation/server.cc

```diff
@@ -82,6 +82,22 @@ void ValidationInstance::initialize(const Options& options,
   InstanceUtil::loadBootstrapConfig(bootstrap_, options,
                                     messageValidationContext().staticValidationVisitor(), *api_);
 
+  // Inject regex engine to singleton.
+  Regex::EnginePtr regex_engine;
+  if (bootstrap_.has_default_regex_engine()) {
+    const auto& default_regex_engine = bootstrap_.default_regex_engine();
+    Regex::EngineFactory& factory =
+        Config::Utility::getAndCheckFactory<Regex::EngineFactory>(default_regex_engine);
+    auto config = Config::Utility::translateAnyToFactoryConfig(
+        default_regex_engine.typed_config(), messageValidationContext().staticValidationVisitor(),
+        factory);
+    regex_engine = factory.createEngine(*config, serverFactoryContext());
+  } else {
+    regex_engine = std::make_shared<Regex::GoogleReEngine>();
+  }
+  Regex::EngineSingleton::clear();
+  Regex::EngineSingleton::initialize(regex_engine.get());
```

Comment:
What do you think of moving this block to a helper function in `source/server`? (For example like `createContextManager` in ssl_context_manager.h? Except that in Server::Instance the result is assigned to a field I think it's identical.

Reply:
Thanks, fixed.